ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -175,23 +177,35 @@ private Response handleStatement(
       final PreparedStatement<?> statement
   )  {
     try {
-      authorizationValidator.checkAuthorization(
-          serviceContext,
-          ksqlEngine.getMetaStore(),
-          statement.getStatement()
-      );
+      final Consumer<KsqlAuthorizationValidator> authValidationConsumer =
+          ksqlAuthorizationValidator -> ksqlAuthorizationValidator.checkAuthorization(
+              serviceContext,
+              ksqlEngine.getMetaStore(),
+              statement.getStatement()
+          );
 
       if (statement.getStatement() instanceof Query) {
         final PreparedStatement<Query> queryStmt = (PreparedStatement<Query>) statement;
 
         if (queryStmt.getStatement().isPullQuery()) {
+          final boolean skipAccessValidation = ksqlConfig.getBoolean(
+              KsqlConfig.KSQL_PULL_QUERIES_SKIP_ACCESS_VALIDATOR_CONFIG);
+          if (authorizationValidator.isPresent() && !skipAccessValidation) {
+            return Errors.badRequest("Pull queries are not currently supported when "
+                + "access validation against Kafka is configured. If you really want to "
+                + "bypass this limitation please set "
```

Comment:
Can you add a warning message or specify the security reasons for disabling the check here?. Users won't see the config doc when seeing this error. They might just go and change this config without knowing the security reasons behind this.

Reply:
We already print the doc of the class also, which calls out what this exposes.